doc/source/serve/tutorials/triton-server-integration.md

```diff
@@ -0,0 +1,167 @@
+
+vae.forward = vae.decode
+torch.onnx.export(
+    vae,
+    (torch.randn(1, 4, 64, 64), False),
+    "vae.onnx",
+    input_names=["latent_sample", "return_dict"],
+    output_names=["sample"],
+    dynamic_axes={
+        "latent_sample": {0: "batch", 1: "channels", 2: "height", 3: "width"},
+    },
+    do_constant_folding=True,
+    opset_version=14,
+)
+
+text_input = tokenizer(
+    prompt,
+    padding="max_length",
+    max_length=tokenizer.model_max_length,
+    truncation=True,
+    return_tensors="pt",
+)
+
+torch.onnx.export(
+    text_encoder,
+    (text_input.input_ids.to(torch.int32)),
+    "encoder.onnx",
+    input_names=["input_ids"],
+    output_names=["last_hidden_state", "pooler_output"],
+    dynamic_axes={
+        "input_ids": {0: "batch", 1: "sequence"},
+    },
+    opset_version=14,
+    do_constant_folding=True,
+)
+```
+
+From the script, the outputs are `vae.onnx` and `encoder.onnx`.
+
+After the ONNX model exported, converting the ONNX model to the TensorRT engine serialized file. ([Details](https://github.com/NVIDIA/TensorRT/blob/release/9.2/samples/trtexec/README.md?plain=1#L22) about trtexec cli)
+```bash
+trtexec --onnx=vae.onnx --saveEngine=vae.plan --minShapes=latent_sample:1x4x64x64 --optShapes=latent_sample:4x4x64x64 --maxShapes=latent_sample:8x4x64x64 --fp16
+```
+
+### Prepare Model Repository
+Triton Server requires a [model repository](https://github.com/triton-inference-server/server/blob/main/docs/user_guide/model_repository.md) to store the models, which is a local directory or remote blob store (e.g. AWS S3) containing the model configuration and the model files.
+In our example, we will use a local directory as the model repository to save all the model files.
+
+```bash
+model_repo/
+├── stable_diffusion
+│   ├── 1
+│   │   └── model.py
+│   └── config.pbtxt
+├── text_encoder
+│   ├── 1
+│   │   └── model.onnx
+│   └── config.pbtxt
+└── vae
+    ├── 1
+    │   └── model.plan
+    └── config.pbtxt
+```
+
+The model repository contains three models: `stable_diffusion`, `text_encoder` and `vae`. Each model has a `config.pbtxt` file and a model file. The `config.pbtxt` file contains the model configuration, which is used to describe the model type and input/output formats. (You can learn more about model config file [here](https://github.com/triton-inference-server/server/blob/main/docs/user_guide/model_configuration.md)). To get config files for our example, you can download them from [here](https://github.com/triton-inference-server/tutorials/tree/main/Conceptual_Guide/Part_6-building_complex_pipelines/model_repository). We use `1` as the version of each model. The model files are saved in the version directory.
+
+
+## Start Ray Serve with the Triton Server
+Triton Server provides python API to start the Triton Server instance. You can use the `nvcr.io/nvidia/tritonserver:23.12-py3` image which already have the Triton Server python API library installed.
```

Comment:
Can we add the complete instructions for how to build an image that has both triton and ray?

Reply:
done.